lib/matplotlib/contour.py

```diff
@@ -124,6 +124,13 @@ def clabel(self, levels=None, *,
             of texts during the drawing time, therefore this can be used if
             aspect of the axes changes.
 
+        zorder : float or None, optional
+            zorder of the contour labels.
+
+            If not specified, the zorder of contour labels is set to either
+            (2 + zorder of contours) or (1 + zorder of contours) depending on
```

Comment:
It is not clear to me where this choice (1+zorder/2+zorder) comes from.

Reply:
https://matplotlib.org/gallery/misc/zorder_demo.html
contourf defaults to zorder of 1 (since filled contours are patch collection), contour defaults to zorder of 2 (line collection). Before this PR clabels always defaulted to zorder of 3 (text object) for both contour and contourf. I am trying to follow the same convention.

Reply:
I'm not really sure it's worth trying to follow an old, clearly "accidental" API?
I think something more reasonable would be e.g. to always use (contour_zorder + 2) which makes sure it goes both above patches and lines, and can be explained relatively easily in the docstring without raising eyebrows.

Reply:
done.